java/ql/lib/ext/kotlin.io.model.yml

```diff
@@ -3,6 +3,7 @@ extensions:
       pack: codeql/java-all
       extensible: sinkModel
     data:
+      - ["kotlin.io", "FilesKt", False, "appendText$default", "(File,String,Charset,int,Object)", "", "Argument[0]", "path-injection", "ai-manual"]
```

Comment:
I believe it's modeling [File.appendText] (the file it's in has an `@file:JvmName("FilesKt")` annotation).

As I understand it, Kotlin generates method names suffixed with `$default` when it is generating methods to handle its default-arguments. (cf [discussion.kotlin.org] and [KU Leven training doc]).

The method in Kotlin is declared 

```kotlin
public fun File.appendText(text: String, charset: Charset = Charsets.UTF_8): Unit =
    FileOutputStream(this, true).use { it.writeTextImpl(text, charset) }
```

So because of the `charset = Charsets.UTF_8` default parameter, the Kotlin compiler generates a `$default` method. The `$default` method generated includes the extra `int` and `Object` because those are related to keeping track of which arguments were (not) passed.

So as I understand it, the following change would be appropriate:

```suggestion
      - ["kotlin.io", "FilesKt", False, "appendText", "", "", "Argument[0]", "path-injection", "ai-manual"]
```

But someone with more Kotlin experience might be better suited to answer this.

cc @igfoo @tamasvajk 

[File.appendText]: https://github.com/JetBrains/kotlin/blob/0938b46726b9c6938df309098316ce741815bb55/libraries/stdlib/jvm/src/kotlin/io/FileReadWrite.kt#L142
[discuss.kotlin.org]: https://discuss.kotlinlang.org/t/help-kotlin-method-generation-problem-in-jax-rs/190/2
[KU Leven training doc]: https://kuleuven-diepenbeek.github.io/appdev-course/lang/kotlin-advanced/

Reply:
Based on the other models in that file, I believe they are still defined on `kotlin.io.FilesKt` (If I understand correctly, Kotlin extensions are similar to C# extension methods in that they are considered static methods of the class that declared them, not as methods on the class they extend).

E.g. `kotlin.io.FilesKt#readText(File,Charset)` which is declared:

```kotlin
@file:JvmName("FilesKt")

public fun File.readText(charset: Charset = Charsets.UTF_8): String = reader(charset).use { it.readText() }
```

Reply:
I've left a comment about this in the codeql-kotlin Slack channel, so hopefully someone from the Kotlin team can confirm if my understanding is accurate.

Reply:
From the Slack discussion it sounds like this may require some more behind-the-scenes work. Shall I take this out for now to maybe reconsider later?

Reply:
That'd probably be the fastest way to unblock this.